core/iwasm/libraries/wasi-nn/test/testapp.c

```diff
@@ -0,0 +1,260 @@
+/*
+ * Copyright (C) 2019 Intel Corporation.  All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <stdint.h>
+
+#include <assert.h>
+#include "wasi_nn.h"
+
+#include <fcntl.h>
+#include <errno.h>
+
+#define BUFFER_SIZE 32
+
+#define INPUT_SIZE 25
+
+#define EPSILON 1e-8
+
+void
+my_load(char *model_name)
+{
+    printf("loading model\n");
+    FILE *pFile = fopen(model_name, "r");
+
+    assert(pFile != NULL);
+
+    uint8_t *buffer;
+    size_t result;
+
+    int lsize = 85000000;
+
+    // allocate memory to contain the whole file:
+    buffer = (uint8_t *)malloc(sizeof(uint8_t) * lsize);
+
+    if (buffer == NULL) {
+        fputs("Memory error\n", stderr);
+        exit(2);
+    }
+
+    result = fread(buffer, 1, lsize, pFile);
+
+    printf("I copied %ld elements\n", result);
+
+    uint32_t *size = malloc(sizeof(uint32_t));
+    *size = 85000000;
+
+    graph_builder_array arr = malloc(sizeof(graph_builder) * 2);
+
+    arr[0] = buffer;
+    arr[1] = (graph_builder)size;
+
+    load(arr, 1);
+
+    printf("loaded model successfully\n");
+    fclose(pFile);
+    free(buffer);
+
+    return;
+}
+
+void
+my_input(float *input_tensor, uint32_t *dim)
+{
+    printf("loading inputs\n");
+
```

Comment:
put this is a different file, we want to split the functions that runs the things of the tests per se.

Reply:
ongoing refractor